client/src/templates/Challenges/components/preview-portal.tsx

```diff
@@ -72,16 +86,17 @@ class PreviewPortal extends Component<PreviewPortalProps> {
       this.props.togglePane('showPreviewPortal');
     });
 
-    this.props.storePortalDocument(this.externalWindow?.document);
-
-    this.mainWindow?.addEventListener('beforeunload', () => {
-      this.externalWindow?.close();
-    });
+    this.props.storePortalWindow(this.externalWindow);
   }
 
   componentWillUnmount() {
-    this.externalWindow?.close();
-    this.props.removePortalDocument();
+    if (
+      this.mainWindow.location.href.split('/').length !== 8 ||
+      this.mainWindow.location.href.split('/')[5] !== 'responsive-web-design'
+    ) {
```

Comment:
The problem with this approach is that it relies on the precise href and that's very likely to change over time.

Since the original issue about the window closing after submission, we can make use of that fact to keep the window open if and only if the user has just submitted a challenge.

This will require a bit more redux shenanigans, we can make use of the completion-epic to solve this. It knows both when the user is submitting and when that submission will move them to a new challenge.  If that dispatches a new action *before* navigating, then we can see if the user `isNavigatingToChallenge` and check that here (instead of looking at the href).

We can also set that property to false here and then all navigation that's *not* triggered by submission will close the window.

Reply:
I can do that for you, (I should point out that we might loose some desired functionality, perhaps it is better user experience for the window to stay open while navigating forward and backwards freely, not just navigation thats triggered by submission.)

regretfully, currently im unable to get the application up and running on my manchine so im unable to implement your request atm

Reply:
It's possible there's a better approach that allows that, but doesn't rely on the path, I don't know.

> im unable to implement your request atm

Don't worry, it's not urgent.

Reply:
> doesn't rely on the path,

I dont see whats wrong with using a path, even the epic is relying on a path, instead of using a hardcoded string value I can just import constant from the superblock, than at least we will still get the keep the functionality of keeping the window open as the user freely navigates and not only based upon submission of a challenge, that wont be good enough to close out the issue? 
 
![image](https://user-images.githubusercontent.com/37171082/204078903-c781dd7c-9d01-40e3-93e8-a549b5763acd.png)

Reply:
It's true that this does rely on the superblock being part of the path, but that's not a great approach.  We have the page context available to determine if a given page is a challenge or not. It's also possible to tell if the next page is a challenge without inspecting the path.

I don't see a good way of allowing the user to navigate anywhere without relying on the details of the path.  We can't constrain ourselves to a precise path structure long term, so we shouldn't rely on that here.

Reply:
it is behvaving as expected and not relying on the path. I also checked the edgecase of the last step before it moves on to the javascript projects, and it closes as expected. Is there any more work to do here?